src/app/pages/component-sidenav/component-sidenav.ts

```diff
@@ -37,17 +40,76 @@ export class ComponentSidenav implements OnInit {
         this.sidenav.close();
       }
     });
+
     // Combine params from all of the path into a single object.
     this.params = Observable.combineLatest(
-      this._route.pathFromRoot.map(route => route.params),
-      Object.assign);
+        this._route.pathFromRoot.map(route => route.params),
+        Object.assign);
   }
 
   isScreenSmall(): boolean {
     return this.mediaMatcher.matches;
   }
 }
 
+@Component({
+  selector: 'app-component-nav',
+  templateUrl: './component-nav.html',
+  animations: [
+    trigger('bodyExpansion', [
+      state('collapsed', style({height: '0px', visibility: 'hidden'})),
+      state('expanded', style({height: '*', visibility: 'visible'})),
+      transition('expanded <=> collapsed', animate('225ms cubic-bezier(0.4,0.0,0.2,1)')),
+    ]),
+  ],
+})
+export class ComponentNav implements OnInit {
+
+  @Input() params: Observable<Params>;
+  expansions = {};
+
+  constructor(public docItems: DocumentationItems,
+              private _router: Router) { }
+
+  ngOnInit() {
+    this._router.events.subscribe(() => this.setExpansions());
```

Comment:
Won't this create a ton of never-completing subscriptions? How about,

```ts
ngOnInit() {
  this._router.events
    .switchMap(() => this.params)
    .takeUntil(this._onDestroy)
    .subscribe(p => this.setExpansions(p));
}

ngOnDestroy() {
  this._onDestroy.next();
  this._onDestroy.complete();
}

setExpansions(p: Params) {
  // ...
}

Reply:
You're right. Thanks for the suggestion!!